src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -518,25 +571,52 @@ void Snippet::generate() {
                     return convert->get_input_element_type(0) != ov::element::f32;
                 return true;
             });
+    schedule = snippet->generate(optManager, reinterpret_cast<const void*>(jcp));
+}
+
+void Snippet::updateSrcDstPtrs(jit_snippets_call_args& call_args) const {
+    for (size_t i = 0; i < srcMemPtrs.size(); i++)
+        call_args.src_ptrs[i] = reinterpret_cast<const uint8_t*>(srcMemPtrs[i]->GetData()) + start_offset_in[i];
 
-    schedule = snippet->generate(optManager, reinterpret_cast<void*>(&jcp));
+    for (size_t i = 0; i < dstMemPtrs.size(); i++)
+        call_args.dst_ptrs[i] = reinterpret_cast<uint8_t*>(dstMemPtrs[i]->GetData()) + start_offset_out[i];
 }
 
-void Snippet::schedule_6d(const jit_snippets_call_args& call_args) const {
+void Snippet::execute(dnnl::stream strm) {
+    if (schedule.ptr == nullptr) {
+        IE_THROW() << "Snippet can't use Optimized implementation and can't fallback to reference";
+    }
+    jit_snippets_call_args call_args;
+    updateSrcDstPtrs(call_args);
+
+    std::vector<jit_snippets_call_args> per_thread_call_args(parallel_get_max_threads(), call_args);
+    if (buffer_scratchpad_size > 0) {
```

Comment:
Not entirely sure why we need to copy jit_snipepts call args nthreads time even if we don't need a buffer. Suggestion: let's keep original signature of `schedule_6d` and create a `schedule_6d_per_thread_buffer` (also with the same signature) that will create a vector of call args inside. We did pretty much the same in the dynamism PR.

Reply:
Added. Thanks!